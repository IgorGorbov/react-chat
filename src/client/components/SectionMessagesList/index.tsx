import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { MessagesList } from '../../styledComponents/index';
import Message from '../Message/index';

interface IMessagesListProps {
  messages: IMessage[];
  currentUser: IUser;
  currentChatId: number;
  selectedMessages: number[];
  selectMessage: (messageId: number) => { type: string; payload: number };
  readMessages: (
    currentChatId: number,
    currentUserID: number,
  ) => {
    type: string;
    payload: { currentChatId: number; currentUserID: number };
  };
}

class SectionMessagesList extends React.Component<IMessagesListProps> {
  private container: any;

  constructor(props: IMessagesListProps) {
    super(props);
  }

  public componentDidMount() {
    this.scrollToBottom();
    // const { currentUser, currentChatId, readMessages } = this.props;
    // readMessages(currentChatId, currentUser.id);
  }

  public componentDidUpdate() {
    this.scrollToBottom();
  }

  public scrollToBottom = () => {
    const messagesContainer: any = ReactDOM.findDOMNode(this.container);
    messagesContainer.scrollTop = messagesContainer.scrollHeight;
  };

  public render() {
    const {
      messages,
      currentUser,
      selectMessage,
      selectedMessages,
    } = this.props;
    return (
      <MessagesList ref={(el: any) => (this.container = el)}>
        {messages.length
          ? messages.map((message: any) => (
              <Message
                key={message.id}
                message={message}
                currentUser={currentUser}
                className="wow bounceIn"
                selectMessage={selectMessage}
                selectedMessages={selectedMessages}
              />
            ))
          : null}
      </MessagesList>
    );
  }
}

export default SectionMessagesList;
