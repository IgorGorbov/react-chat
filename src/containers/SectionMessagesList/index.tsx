import * as React from 'react';
import {
  SectionMessagesList,
  UserAvatar,
  Text,
  MessageContainer,
  MessageInfoContainer,
  MessageArrow,
  MessageDate,
} from '../../StyledComponents';

const src = 'https://emilcarlsson.se/assets/mikeross.png';

class MessagesList extends React.Component {
  public static messageRender(massage: IMessage): JSX.Element {
    return (
      <MessageContainer key={Math.random()}>
        <UserAvatar
          maxwidth="45px"
          maxheight="45px"
          bordersize="1px"
          src={src}
          size="tiny"
          circular={true}
        />
        <MessageInfoContainer>
          <MessageArrow />
          <Text
            textColor="white"
            textSize="15px"
            textLineHeight="1.5"
            textWeight="400"
            textFontFamily="Titillium Web"
            className="messageText"
          >
            {massage.text}
          </Text>
          <MessageDate>{massage.date}</MessageDate>
        </MessageInfoContainer>
      </MessageContainer>
    );
  }
  public render() {
    const messages: IMessage[] = [
      {
        from: 'Mike Ross',
        to: 'Louis Litt',
        text:
          'Lorem ipsum dolor sit amet, onsectetur adipisicing  onsectetur ' +
          'adipisicing consectetur adipisicing elit. Porro!',
        date: new Date().toDateString(),
        isRead: false,
      },
      {
        from: 'Louis Litt',
        to: 'Mike Ross',
        text: 'Lorem ipsum dolor sit amet',
        date: new Date().toDateString(),
        isRead: false,
      },
      {
        from: 'Mike Ross',
        to: 'Louis Litt',
        text:
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Porro!',
        date: new Date().toDateString(),
        isRead: false,
      },
      {
        from: 'Louis Litt',
        to: 'Mike Ross',
        text: 'Lorem ipsum dolor sit amet',
        date: new Date().toDateString(),
        isRead: false,
      },
      {
        from: 'Mike Ross',
        to: 'Louis Litt',
        text:
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Porro!',
        date: new Date().toDateString(),
        isRead: false,
      },
      {
        from: 'Louis Litt',
        to: 'Mike Ross',
        text: 'Lorem ipsum dolor sit amet',
        date: new Date().toDateString(),
        isRead: false,
      },
      {
        from: 'Mike Ross',
        to: 'Louis Litt',
        text:
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Porro!',
        date: new Date().toDateString(),
        isRead: false,
      },
      {
        from: 'Louis Litt',
        to: 'Mike Ross',
        text: 'Lorem ipsum dolor sit amet',
        date: new Date().toDateString(),
        isRead: false,
      },
      {
        from: 'Mike Ross',
        to: 'Louis Litt',
        text:
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Porro!',
        date: new Date().toDateString(),
        isRead: false,
      },
      {
        from: 'Louis Litt',
        to: 'Mike Ross',
        text: 'Lorem ipsum dolor sit amet',
        date: new Date().toDateString(),
        isRead: false,
      },
    ];

    return (
      <SectionMessagesList>
        {messages.map(MessagesList.messageRender)}
      </SectionMessagesList>
    );
  }
}

export default MessagesList;
