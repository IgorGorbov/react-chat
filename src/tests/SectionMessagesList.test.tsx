import * as React from "react";
import toJSON from 'enzyme-to-json';
import { shallow, mount } from "enzyme";
import SectionMessagesList from "../containers/SectionMessagesList";
import {
  MessageContainer,
  UserAvatar,
  MessageInfoContainer,
  MessageArrow,
  Text,
  MessageDate
} from '../StyledComponents'

describe('testing SectionMessagesList', () => {
  const messages: IMessage[] = [
    {
      id: '1',
      from: 'Mike Ross',
      to: 'Louis Litt',
      text:
      'Lorem ipsum dolor sit amet, onsectetur adipisicing  onsectetur ' +
      'adipisicing consectetur adipisicing elit. Porro!',
      date: new Date().toDateString(),
      isRead: false,
    },
    {
      id: '2',
      from: 'Mike Jenn',
      to: 'Louis Litt',
      text:
      'Lorem ipsum dolor sit amet!',
      date: new Date().toDateString(),
      isRead: false,
    }
  ];
  const wrapper = mount(<SectionMessagesList messages={messages} />);

  it("should contain SectionMessagesList", () => {
    expect(wrapper.find('SectionMessagesList')).toHaveLength(1);
  });

  it("should render 0 messages", () => {
    const result = mount(<SectionMessagesList messages={[]} />);
    expect(result.find('p').text()).toEqual('Not messages');
  });

  it("should contain 2 messages", () => {
    expect(wrapper.find(MessageContainer)).toHaveLength(2);
  });

  it("should contain 2 UserAvatars", () => {
    expect(wrapper.find(UserAvatar)).toHaveLength(2);
  });

  it("should contain 2 MessageInfoContainer", () => {
    expect(wrapper.find(MessageInfoContainer)).toHaveLength(2);
  });

  it("should contain 2 MessageArrow", () => {
    expect(wrapper.find(MessageArrow)).toHaveLength(2);
  });

  it("should contain 2 Messages Text", () => {
    expect(wrapper.find(Text)).toHaveLength(2);
  });

  it("should the text of the messages contain fake data", () => {
    wrapper.find(Text).forEach((node, i) => {
      expect(node.text()).toEqual(messages[i].text)
    })
  });

  it("should contain 2 Messages MessageDate", () => {
    expect(wrapper.find(MessageDate)).toHaveLength(2);
  });

  it("should the data of the messages contain fake data", () => {
    wrapper.find(MessageDate).forEach((node, i) => {
      expect(node.text()).toEqual(messages[i].date)
    })
  });


  it('snapshot SectionMessagesList', () => {
    const result = shallow(<SectionMessagesList messages={messages} />);
    const tree = toJSON(result);
    expect(tree).toMatchSnapshot();
  })
});
