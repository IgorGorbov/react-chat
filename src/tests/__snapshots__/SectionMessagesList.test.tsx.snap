// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing SectionMessagesList snapshot SectionMessagesList 1`] = `
<SectionMessagesList
  currentChatId={1}
  currentUser={
    Object {
      "avatar": "currentAvatar",
      "id": 1,
      "name": "currentName",
      "status": "Online",
    }
  }
  messages={
    Array [
      Object {
        "date": "15:00",
        "deletedBy": Array [],
        "from": Object {
          "avatar": "currentAvatar",
          "id": 1,
          "name": "currentName",
          "status": "Online",
        },
        "id": 1,
        "isRead": false,
        "text": "text test 1",
        "to": Object {
          "avatar": "companionAvatar",
          "id": 2,
          "name": "companionName",
          "status": "Online",
        },
      },
      Object {
        "date": "16:00",
        "deletedBy": Array [],
        "from": Object {
          "avatar": "currentAvatar",
          "id": 1,
          "name": "currentName",
          "status": "Online",
        },
        "id": 2,
        "isRead": false,
        "text": "text test 2",
        "to": Object {
          "avatar": "companionAvatar",
          "id": 2,
          "name": "companionName",
          "status": "Online",
        },
      },
    ]
  }
  readMessages={[MockFunction]}
  selectMessage={
    [MockFunction] {
      "calls": Array [
        Array [
          1,
        ],
      ],
    }
  }
  selectedMessages={Array []}
>
  <styled.section>
    <section
      className="sc-iwsKbI goWDXY"
    >
      <Message
        className="wow bounceIn"
        currentUser={
          Object {
            "avatar": "currentAvatar",
            "id": 1,
            "name": "currentName",
            "status": "Online",
          }
        }
        key="1"
        message={
          Object {
            "date": "15:00",
            "deletedBy": Array [],
            "from": Object {
              "avatar": "currentAvatar",
              "id": 1,
              "name": "currentName",
              "status": "Online",
            },
            "id": 1,
            "isRead": false,
            "text": "text test 1",
            "to": Object {
              "avatar": "companionAvatar",
              "id": 2,
              "name": "companionName",
              "status": "Online",
            },
          }
        }
        selectMessage={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
          }
        }
        selectedMessages={Array []}
      >
        <ReactWOW
          animateClass="animated"
          animation="fadeInRight"
          callback={[Function]}
          delay=""
          disabled={false}
          duration=""
          iteration=""
          offset={0}
          overflow={false}
          resize={true}
          scroll={true}
        >
          <styled.div
            className="isMessageTo fadeInRight"
            key=".$1"
            onAnimationEnd={[Function]}
            onClick={[Function]}
            style={
              Object {
                "animationName": "none",
                "visibility": "hidden",
              }
            }
          >
            <div
              className="isMessageTo fadeInRight sc-gqjmRU fEkjPM"
              onAnimationEnd={[Function]}
              onClick={[Function]}
              style={
                Object {
                  "animationName": "none",
                  "visibility": "hidden",
                }
              }
            >
              <Styled(Image)
                bordersize="1px"
                circular={true}
                maxheight="45px"
                maxwidth="45px"
                size="tiny"
                src="currentAvatar"
              >
                <Image
                  as="img"
                  bordersize="1px"
                  circular={true}
                  className="sc-ifAKCX JhXmE"
                  maxheight="45px"
                  maxwidth="45px"
                  size="tiny"
                  src="currentAvatar"
                  ui={true}
                >
                  <img
                    bordersize="1px"
                    className="ui tiny circular image sc-ifAKCX JhXmE"
                    maxheight="45px"
                    maxwidth="45px"
                    src="currentAvatar"
                  />
                </Image>
              </Styled(Image)>
              <styled.div
                className="isMessageTo"
              >
                <div
                  className="isMessageTo sc-VigVT iTxXta"
                >
                  <styled.span
                    className="isMessageTo"
                  >
                    <span
                      className="isMessageTo sc-jTzLTM jajjDS"
                    />
                  </styled.span>
                  <styled.p>
                    <p
                      className="sc-fjdhpX gAKYZH"
                    >
                      text test 1
                    </p>
                  </styled.p>
                  <styled.span>
                    <span
                      className="sc-jzJRlG dEnjoI"
                    >
                      15:00
                    </span>
                  </styled.span>
                </div>
              </styled.div>
            </div>
          </styled.div>
        </ReactWOW>
      </Message>
      <Message
        className="wow bounceIn"
        currentUser={
          Object {
            "avatar": "currentAvatar",
            "id": 1,
            "name": "currentName",
            "status": "Online",
          }
        }
        key="2"
        message={
          Object {
            "date": "16:00",
            "deletedBy": Array [],
            "from": Object {
              "avatar": "currentAvatar",
              "id": 1,
              "name": "currentName",
              "status": "Online",
            },
            "id": 2,
            "isRead": false,
            "text": "text test 2",
            "to": Object {
              "avatar": "companionAvatar",
              "id": 2,
              "name": "companionName",
              "status": "Online",
            },
          }
        }
        selectMessage={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
          }
        }
        selectedMessages={Array []}
      >
        <ReactWOW
          animateClass="animated"
          animation="fadeInRight"
          callback={[Function]}
          delay=""
          disabled={false}
          duration=""
          iteration=""
          offset={0}
          overflow={false}
          resize={true}
          scroll={true}
        >
          <styled.div
            className="isMessageTo fadeInRight"
            key=".$2"
            onAnimationEnd={[Function]}
            onClick={[Function]}
            style={
              Object {
                "animationName": "none",
                "visibility": "hidden",
              }
            }
          >
            <div
              className="isMessageTo fadeInRight sc-gqjmRU fEkjPM"
              onAnimationEnd={[Function]}
              onClick={[Function]}
              style={
                Object {
                  "animationName": "none",
                  "visibility": "hidden",
                }
              }
            >
              <Styled(Image)
                bordersize="1px"
                circular={true}
                maxheight="45px"
                maxwidth="45px"
                size="tiny"
                src="currentAvatar"
              >
                <Image
                  as="img"
                  bordersize="1px"
                  circular={true}
                  className="sc-ifAKCX JhXmE"
                  maxheight="45px"
                  maxwidth="45px"
                  size="tiny"
                  src="currentAvatar"
                  ui={true}
                >
                  <img
                    bordersize="1px"
                    className="ui tiny circular image sc-ifAKCX JhXmE"
                    maxheight="45px"
                    maxwidth="45px"
                    src="currentAvatar"
                  />
                </Image>
              </Styled(Image)>
              <styled.div
                className="isMessageTo"
              >
                <div
                  className="isMessageTo sc-VigVT iTxXta"
                >
                  <styled.span
                    className="isMessageTo"
                  >
                    <span
                      className="isMessageTo sc-jTzLTM jajjDS"
                    />
                  </styled.span>
                  <styled.p>
                    <p
                      className="sc-fjdhpX gAKYZH"
                    >
                      text test 2
                    </p>
                  </styled.p>
                  <styled.span>
                    <span
                      className="sc-jzJRlG dEnjoI"
                    >
                      16:00
                    </span>
                  </styled.span>
                </div>
              </styled.div>
            </div>
          </styled.div>
        </ReactWOW>
      </Message>
    </section>
  </styled.section>
</SectionMessagesList>
`;
